src/darwin/Framework/CHIP/MTRDevice+XPC.h

```diff
@@ -0,0 +1,47 @@
+//
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Matter/Matter.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MTRDeviceXPCServerProtocol <NSObject>
+
+
+typedef void (^MTRDeviceXPCServerPingResult)(NSString *);
+typedef void (^MTRDeviceXPCServerAttributeReadResult)(NSDictionary<NSString *, id> * _Nullable);
+
+- (void)pingWithResultHandler:(MTRDeviceXPCServerPingResult)resultHandler;
+
+- (void)readAttributeWithEndpointID:(NSNumber *)endpointID
+                          clusterID:(NSNumber *)clusterID
+                        attributeID:(NSNumber *)attributeID
+                             params:(MTRReadParams * _Nullable)params
+                      resultHandler:(MTRDeviceXPCServerAttributeReadResult)resultHandler;
+
```

Comment:
I'd also add a method that has a -(BOOL)returningSomething;  signature to test that path too.